File: scripts/utils/read_history.py
import subprocess
import os

cwd = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
try:
    print("--- Git Status ---")
    res = subprocess.run(["git", "status"], cwd=cwd, capture_output=True, text=True, check=True)
    print(res.stdout)

    print("--- Git Log ---")
    res2 = subprocess.run(["git", "log", "-n", "10", "--oneline", "functions/ashen_api/index.js"], cwd=cwd, capture_output=True, text=True, check=True)
    print(res2.stdout)
except Exception as e:
    print("Error:", e)

File: scripts/utils/read_history.js
const { execSync } = require('child_process');
const path = require('path');
const rootDir = path.join(__dirname, '..', '..');

try {
  console.log("--- Git status ---");
  console.log(execSync('git status', { cwd: rootDir }).toString());
  console.log("--- Git Log of index.js changes ---");
  console.log(execSync('git log -p -n 3 functions/ashen_api/index.js', { cwd: rootDir }).toString());
} catch (e) {
  console.error(e.message);
}
